src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -0,0 +1,282 @@
+                } else {
+                    // LLM sometimes returns the tensor results as a json object directly instead of
+                    // string response, and the json object is stored as a map.
+                    response = StringUtils.toJson(dataMap);
+                }
+                if (!isJson(response)) {
+                    throw new IllegalArgumentException(
+                        LoggerMessageFormat.format(null, "The response from LLM is not a json: [{}]", response)
+                    );
+                }
+                listener.onResponse((T) response);
+            }, e -> {
+                log.error("Failed to run model " + modelId, e);
+                listener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            log.error("Failed to call CreateAlertTool", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private String getIndexMappingInfo(Map<String, String> parameters) throws InterruptedException, ExecutionException {
+        if (!parameters.containsKey("indices") || parameters.get("indices").isEmpty()) {
+            throw new IllegalArgumentException(
+                "No indices in the input parameter. Ask user to "
+                    + "provide index as your final answer directly without using any other tools"
+            );
+        }
+        String rawIndex = parameters.getOrDefault("indices", "");
+        List<String> indexList;
+        try {
+            indexList = gson.fromJson(rawIndex, new TypeToken<List<String>>() {
+            }.getType());
+        } catch (Exception e) {
+            // LLM sometimes returns the indices as a string instead of a json list, although we require that in the tool description.
+            indexList = Collections.singletonList(rawIndex);
+        }
+        if (indexList.isEmpty()) {
+            throw new IllegalArgumentException(
+                "The input indices is empty. Ask user to " + "provide index as your final answer directly without using any other tools"
+            );
+        } else if (indexList.stream().anyMatch(index -> index.startsWith("."))) {
+            throw new IllegalArgumentException(
+                LoggerMessageFormat
+                    .format(
+                        null,
+                        "The provided indices [{}] contains system index, which is not allowed. Ask user to "
+                            + "check the provided indices as your final answer without using any other.",
+                        rawIndex
+                    )
+            );
+        }
+        final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
+        final GetIndexRequest getIndexRequest = new GetIndexRequest()
+            .indices(indices)
+            .indicesOptions(IndicesOptions.strictExpand())
+            .local(Boolean.parseBoolean(parameters.getOrDefault("local", "true")))
+            .clusterManagerNodeTimeout(DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT);
```

Comment:
It's not preferred to blocking wait in code, unless you can confirm it's not running in event loop thread, also, what's the value of DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT?

Reply:
Sorry, what do you mean of blocking wait? I just imitate what IndexMapping do here.

https://github.com/opensearch-project/ml-commons/blob/5ad26df7dd23a6ed37a13d80548779579521b13f/ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java#L139

DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT is 30s.

Reply:
Line 205: client.admin().indices().getIndex(getIndexRequest)`.get()`, this get blocks the thread until result returns, it's preferred to use `getIndex(getIndexRequest, actionListener)` method instead of `getIndex(getIndexRequest)`.

Reply:
I see. But it does need to retrieve index mapping info before requesting LLM. if using `getIndex(getIndexRequest, actionListener)`, then all logic after that should be put in the listener and wait for response from getIndex, that makes not difference with current implementation.